lib/modules/nightMode.js

```diff
@@ -251,14 +259,34 @@ function overrideNightMode() {
 	nightmodeOverrideStorage.set(Date.now());
 }
 
-function isTimeForNightMode() {
+function getGeolocation() {
+	return new Promise(resolve => {
```

Comment:
To avoid freaking out users, we should show a notification/modal explaining why geolocation is about to be requested before/while asking for permission.

Might need to be "during" to avoid showing explanation if user has already granted permission.